src/ensembl/src/content/app/entity-viewer/hooks/useEntityViewerAnalytics.ts

```diff
@@ -56,8 +57,53 @@ const useEntityViewerAnalytics = () => {
     });
   };
 
+  const trackProteinInfoToggle = (params: {
+    transcriptQuality: string | null;
+    transcriptId: string;
+    action: 'open_accordion' | 'close_accordion';
+    transcriptPosition: number;
+  }) => {
+    const { transcriptId, transcriptQuality } = params;
+    const label = transcriptQuality
+      ? `${transcriptQuality} ${transcriptId}` // "MANE Plus Clinical ENST00000380152.8"
+      : transcriptId;
+    analyticsTracking.trackEvent({
+      category: 'gene_view_proteins_list',
+      label,
+      action: params.action,
+      value: params.transcriptPosition + 1
+    });
+  };
+
+  const trackProteinDownload = (params: {
```

Comment:
Can this be generic? I need to use the same for transcript list download, the only change is the category.